.github/check-workflow-write-permission.sh

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script checks if the current user is allowed to change to GitHub Actions workflows. The script should be
+# installed on self-hosted runners and defined in the `ACTIONS_RUNNER_HOOK_JOB_STARTED` environment variable.
+# The script is triggered when a job has been assigned to a runner, but before the job starts running.
+# https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job#triggering-the-scripts
+
+if [[ $GITHUB_REF != refs/pull* ]]; then
+  echo "Not a pull request. Skipping"
+  exit 0
+fi
+
+PR_NUMBER=$(echo "$GITHUB_REF" | awk -F / '{print $3}')
+# To obtain sufficient quota for gh cli, it is recommended to set PAT in the `GH_TOKEN` environment variable.
+# Check if there are any changes in .github/actions or .github/workflows
+WORKFLOW_CHANGES=$(gh -R line/armeria pr diff "$PR_NUMBER" --name-only | grep -c '^.github/workflows\|^.github/actions')
```

Comment:
Question) Is it possible for external contributor change this script itself (which isn't in `workflows`, `actions`)

Reply:
The file is not directly accessed by GitHub Actions. It was added for version control. If there are any changes to this file, we must manually update the file in self-hosted runners.